### ISSUE TITLE
Retrofit 2 - Dynamic Error Message Parsing Issue

### ISSUE THREAD

**Issue by bhagyasri** [square/retrofit#2073]
I have used Retrofit 2 for API calls in my app. I have difficulty to parse Retrofit error response "response.errorBody()". Because some times I am getting responseMessage as object and some times as string or json array. How to rectify this to parse dynamic error response from API?

**Comment by JakeWharton** [comment]
You can't rely on the format of an error body because it may be returned by your application server (in the case of 400s), your server software (in the case of 500s), or an intermediate proxy (any response code). Retrofit exposes the unprocessed error body for this reason.

If you want, you can add your own implementation of `Call` (or whatever type you want) via a `CallAdapter.Factory` that exposes the error body in whatever fashion you want. I used this as an example recently in a talk I gave: https://speakerdeck.com/jakewharton/making-retrofit-work-for-you-droidcon-uk-2016?slide=217. There's no associated code, you would have to write that yourself.

The bottom line is really in the first paragraph. There's little that can be done because the contents of a non-200 response can never be guaranteed. This is why Retrofit exposes it as raw bytes and will continue to do so.
